src/main/resources/com/google/api/codegen/csharp/gapic_client.snip

```diff
@@ -108,41 +120,67 @@
+                /// <item><description>Retry maximum delay: {@call.retryParamsView.maxRetryDelay.getMillis} milliseconds</description></item>
+                /// <item><description>Initial timeout: {@call.retryParamsView.initialRpcTimeout.getMillis} milliseconds</description></item>
+                /// <item><description>Timeout multiplier: {@call.retryParamsView.rpcTimeoutMultiplier}</description></item>
+                /// <item><description>Timeout maximum delay: {@call.retryParamsView.maxRpcTimeout.getMillis} milliseconds</description></item>
+                /// </list>
+                /// Retry will be attempted on the following response status codes:
+                /// <list>
+                @if call.retryCodesView.codeNames
+                    @join code : call.retryCodesView.codeNames
+                        /// <item><description><see cref="StatusCode.{@code}"/></description></item>
+                    @end
+                @else
+                    /// <item><description>No status codes</description></item>
                 @end
-            @else
-                /// <item><description>No status codes</description></item>
+                /// </list>
+                /// Default RPC expiration is {@call.retryParamsView.totalTimeout.getMillis} milliseconds.
+                /// </remarks>
+                public CallSettings {@call.memberName} { get; set; } = CallSettings.FromCallTiming(
+                    CallTiming.FromRetry(new RetrySettings(
+                        retryBackoff: {@call.retryParamsView.retryBackoffMethodName}(),
+                        timeoutBackoff: {@call.retryParamsView.timeoutBackoffMethodName}(),
+                        totalExpiration: Expiration.FromTimeout(TimeSpan.FromMilliseconds({@call.retryParamsView.totalTimeout.getMillis})),
+                        retryFilter: {@call.retryCodesView.retryFilterMethodName}
+                    )));
+            @case "BidiStreaming"
+                /// <summary>
+                /// <see cref="CallSettings"/> for calls to <c>{@api.name}.{@call.methodName}</c>.
+                /// </summary>
+                /// <remarks>
+                /// Default RPC expiration is {@call.retryParamsView.totalTimeout.getMillis} milliseconds.
+                /// </remarks>
+                public CallSettings {@call.memberName} { get; set; } = CallSettings.FromCallTiming(
+                    CallTiming.FromTimeout(TimeSpan.FromMilliseconds({@call.retryParamsView.totalTimeout.getMillis})));
+
+                /// <summary>
+                /// <see cref="BidirectionalStreamingSettings"/> for calls to
+                /// <c>{@api.name}.{@call.methodName}</c>.
+                /// </summary>
+                /// <remarks>
+                /// The default local send queue size is 100.
+                /// </remarks>
+                public BidirectionalStreamingSettings {@call.methodName}StreamingSettings { get; set; } =
```

Comment:
Presumably we want to parameterize this in the YAML. I think it's worth at least raising an issue so we can remember to do so.